src/onegov/org/forms/page.py

```diff
@@ -55,3 +57,66 @@ class PageUrlForm(ChangeAdjacencyListUrlForm):
 
     def get_model(self):
         return self.model.page
+
+
+class MovePageForm(Form):
+    """ Form to move a page including its subpages. """
+
+    parent_id = ChosenSelectField(
+        label=_("Destination"),
+        choices=[],
+        validators=[
+            InputRequired()
+        ]
+    )
+
+    def on_request(self):
+        pages = PageCollection(self.request.session)
+        self.iterate_page_tree(pages.roots, indent='',
+                               page_list=self.parent_id.choices)
+
+        # adding root element
+        self.parent_id.choices.insert(
+            0, ('root', self.request.translate(_("- Root -")))
+        )
+
+    def iterate_page_tree(self, pages, indent='', page_list=None):
+        """
+        Iterates over the page tree and lists the elements with ident
+        to show the page hierarchy in the choices list
+        """
+        for page in pages:
+            item = (str(page.id), f'{indent} {page.title}')
+            page_list.append(item)
+
+            self.iterate_page_tree(page.children, indent=indent + ' -',
+                                   page_list=page_list)
+
+    def ensure_valid_parent(self):
+        """
+        As a new destination (parent page) every menu item is valid except
+        yourself. You cannot assign yourself as the new destination
+        """
+        if self.parent_id.data and self.parent_id.data.isdigit():
+            new_parent_id = int(self.parent_id.data)
+            page_id_from_url = int(self.request.url.rsplit('/')[-1])
```

Comment:
Try not to ever use request.url since this might lead to unexcepted behaviour (do to X_VHM_HOST etc.). Rather use request.link in general. Here, you might use `self.model`

Reply:
I found a way to read from the model instead of the url